File: test_terminal_display.py
#!/usr/bin/env python3
"""
Test script for terminal display functionality
"""

import sys
import os
sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))

from offchain.core.terminal_display import get_terminal_display
import time

def test_terminal_display():
    """Test all terminal display features"""
    
    display = get_terminal_display()
    
    print("\n" + "="*80)
    print("Testing Terminal Display Module")
    print("="*80 + "\n")
    
    # Test 1: Header
    print("Test 1: Header Display")
    display.print_header("PAPER")
    time.sleep(1)
    
    # Test 2: Opportunity Scanning
    print("\nTest 2: Opportunity Scanning")
    display.log_opportunity_scan(
        token="USDC",
        chain_id=137,
        dex1="UNIV3",
        dex2="SUSHI",
        amount_usd=1000,
        profitable=False,
        gas_gwei=25.5
    )
    time.sleep(0.5)
    
    display.log_opportunity_scan(
        token="WETH",
        chain_id=1,
        dex1="UNIV3",
        dex2="SUSHI",
        amount_usd=2000,
        profitable=True,
        profit_usd=15.50,
        gas_gwei=45.2,
        details="Size: $2000"
    )
    time.sleep(1)
    
    # Test 3: Decision Logging
    print("\nTest 3: Decision Logging")
    display.log_decision(
        decision_type="AI_TUNE",
        token="USDC",
        chain_id=137,
        reason="AI-optimized execution parameters",
        details={'slippage': 50, 'priority': 30}
    )
    time.sleep(0.5)
    
    display.log_decision(
        decision_type="APPROVE",
        token="WETH",
        chain_id=1,
        reason="Profitable opportunity approved",
        details={'profit_usd': 15.50, 'gas_gwei': 45.2}
    )
    time.sleep(1)
    
    # Test 4: Signal Generation
    print("\nTest 4: Signal Generation")
    display.log_signal_generated(
        token="WETH",
        chain_id=1,
        profit_usd=15.50,
        route=["UNIV3", "SUSHI"],
        gas_gwei=45.2,
        execution_params={'slippage': 50, 'priority': 30}
    )
    time.sleep(1)
    
    # Test 5: Gas Updates
    print("\nTest 5: Gas Price Updates")
    display.log_gas_update(chain_id=137, gas_gwei=25.5, threshold=50.0)
    display.log_gas_update(chain_id=1, gas_gwei=150.0, threshold=100.0)
    time.sleep(1)
    
    # Test 6: Stats Bar
    print("\nTest 6: Stats Bar")
    # Simulate some activity
    display.stats['opportunities_scanned'] = 150
    display.stats['opportunities_profitable'] = 12
    display.stats['opportunities_signaled'] = 8
    display.stats['executions_attempted'] = 8
    display.stats['executions_successful'] = 7
    display.stats['executions_failed'] = 1
    display.stats['total_profit_usd'] = 125.50
    display.print_stats_bar()
    time.sleep(1)
    
    # Test 7: Warnings and Errors
    print("\nTest 7: Warnings and Errors")
    display.log_warning("BRAIN", "Gas price approaching threshold")
    display.log_error("BRAIN", "Failed to fetch price", "API rate limit exceeded")
    time.sleep(1)
    
    # Test 8: Info messages
    print("\nTest 8: Info Messages")
    display.log_info("System initialized successfully")
    time.sleep(1)
    
    print("\n" + "="*80)
    print("✅ All Terminal Display Tests Passed!")
    print("="*80 + "\n")

if __name__ == "__main__":
    test_terminal_display()

File: test_terminal_display.js
#!/usr/bin/env node
/**
 * Test script for JavaScript terminal display functionality
 */

const path = require('path');
const { terminalDisplay } = require(path.join(__dirname, 'offchain', 'execution', 'terminal_display.js'));

function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

async function testTerminalDisplay() {
    console.log('\n' + '='.repeat(80));
    console.log('Testing JavaScript Terminal Display Module');
    console.log('='.repeat(80) + '\n');

    // Test 1: Header
    console.log('Test 1: Header Display');
    terminalDisplay.printHeader('PAPER');
    await sleep(1000);

    // Test 2: Signal Reception
    console.log('\nTest 2: Signal Reception');
    terminalDisplay.logSignalReceived('signal_123', 'USDC', 137, 12.50);
    await sleep(500);

    // Test 3: Execution Start
    console.log('\nTest 3: Execution Start');
    terminalDisplay.logExecutionStart('PAPER-1-123456', 'USDC', 137, '1000000000', 'PAPER');
    await sleep(500);

    // Test 4: Gas Estimate
    console.log('\nTest 4: Gas Estimate');
    terminalDisplay.logGasEstimate(137, '30', 5.50, 12.50);
    await sleep(500);

    // Test 5: Decision
    console.log('\nTest 5: Decision Logging');
    terminalDisplay.logDecision('APPROVE', 'USDC', 137, 'Profitable trade', { profit: 12.50, gas: 30 });
    await sleep(500);

    // Test 6: Execution Complete
    console.log('\nTest 6: Execution Complete');
    terminalDisplay.logExecutionComplete('PAPER-1-123456', 'SIMULATED', 150, 12.50);
    await sleep(1000);

    // Test 7: Stats Bar
    console.log('\nTest 7: Stats Bar');
    // Simulate some activity
    terminalDisplay.stats.signalsProcessed = 25;
    terminalDisplay.stats.executionsAttempted = 20;
    terminalDisplay.stats.executionsSuccessful = 18;
    terminalDisplay.stats.executionsFailed = 2;
    terminalDisplay.stats.paperTrades = 20;
    terminalDisplay.stats.totalProfitUSD = 250.75;
    terminalDisplay.printStatsBar();
    await sleep(1000);

    // Test 8: Warnings and Errors
    console.log('\nTest 8: Warnings and Errors');
    terminalDisplay.logWarning('BOT', 'Low balance detected');
    terminalDisplay.logError('BOT', 'Failed to execute', 'Insufficient gas');
    await sleep(1000);

    // Test 9: Info messages
    console.log('\nTest 9: Info Messages');
    terminalDisplay.logInfo('Execution engine started successfully');
    await sleep(1000);

    console.log('\n' + '='.repeat(80));
    console.log('✅ All JavaScript Terminal Display Tests Passed!');
    console.log('='.repeat(80) + '\n');
}

testTerminalDisplay().catch(err => {
    console.error('Test failed:', err);
    process.exit(1);
});
